testng-core/src/main/java/org/testng/internal/NoOpTestClass.java

```diff
@@ -37,52 +39,52 @@ protected NoOpTestClass() {
 
   public NoOpTestClass(ITestClass testClass) {
     m_testClass = testClass.getRealClass();
-    m_beforeSuiteMethods = testClass.getBeforeSuiteMethods();
-    m_beforeTestConfMethods = testClass.getBeforeTestConfigurationMethods();
+    m_beforeSuiteMethods = List.of(testClass.getBeforeSuiteMethods());
+    m_beforeTestConfMethods = List.of(testClass.getBeforeTestConfigurationMethods());
     m_beforeGroupsMethods = testClass.getBeforeGroupsMethods();
-    m_beforeClassMethods = testClass.getBeforeClassMethods();
-    m_beforeTestMethods = testClass.getBeforeTestMethods();
-    m_afterSuiteMethods = testClass.getAfterSuiteMethods();
-    m_afterTestConfMethods = testClass.getAfterTestConfigurationMethods();
-    m_afterGroupsMethods = testClass.getAfterGroupsMethods();
-    m_afterClassMethods = testClass.getAfterClassMethods();
-    m_afterTestMethods = testClass.getAfterTestMethods();
+    m_beforeClassMethods = List.of(testClass.getBeforeClassMethods());
+    m_beforeTestMethods = List.of(testClass.getBeforeTestMethods());
+    m_afterSuiteMethods = List.of(testClass.getAfterSuiteMethods());
+    m_afterTestConfMethods = List.of(testClass.getAfterTestConfigurationMethods());
+    m_afterGroupsMethods = List.of(testClass.getAfterGroupsMethods());
+    m_afterClassMethods = List.of(testClass.getAfterClassMethods());
+    m_afterTestMethods = List.of(testClass.getAfterTestMethods());
     m_instances = IObject.objects(testClass, true);
     m_instanceHashes = IObject.instanceHashCodes(testClass);
     m_xmlTest = testClass.getXmlTest();
     m_xmlClass = testClass.getXmlClass();
   }
 
   public void setBeforeTestMethods(ITestNGMethod[] beforeTestMethods) {
-    m_beforeTestMethods = beforeTestMethods;
+    m_beforeTestMethods = List.of(beforeTestMethods);
   }
 
   public void setAfterTestMethod(ITestNGMethod[] afterTestMethods) {
```

Comment:
The method signature can be updated here because it is not part of api and the class is located into internal

Reply:
I will prefer to leave this as is, because this is going to just add more changes elsewhere. The caller of this method works with an array and so it will still need to be changed to a List in multiple places. Instead its a lot more simpler if the translation from array to list happens just in one place.